CarpetLib/src/data.cc

```diff
@@ -659,6 +659,64 @@ void data<T>::transfer_prolongate(data const *const src, ibbox const &dstbox,
     break;
   }
 
+  case op_Lagrange_third_order_prolong: {
+    static Timer timer("prolongate_Lagrange_3o_prolong");
+    timer.start();
+    // enum centering { vertex_centered, cell_centered };
+    switch (cent) {
+    case vertex_centered: {
+      static void (*the_operators[])(
+          T const *restrict const src, ivect3 const &restrict srcpadext,
+          ivect3 const &restrict srcext, T *restrict const dst,
+          ivect3 const &restrict dstpadext, ivect3 const &restrict dstext,
+          ibbox3 const &restrict srcbbox, ibbox3 const &restrict dstbbox,
```

Comment:
@eschnett says

This chooses third order interpolation even if the overall order is set to be first order. This will bypass the check for sufficient number of ghost zones. The operator’s check will still catch this, but with an error message that is difficult to interpret.

I suggest changing this, so that the `the_operators` array is set up to contain the actual desired operators, and then using the respective array element here.

Otherwise there is no point in setting up an array, and you can simple pass `&prolongate_3d_rf2<T, 3>` as argument here.